pkg/dcmtime/common.go

```diff
@@ -0,0 +1,147 @@
+package dcmtime
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// isIncluded returns whether `check` is included in `limit`.
+//
+// Example: to test whether seconds should be included, you would:
+// isIncluded(PrecisionSeconds, [caller-passed-limit])
+func isIncluded(check PrecisionLevel, precision PrecisionLevel) bool {
+	return check <= precision
+}
+
+// truncateMilliseconds truncate nanosecond time.Time value to arbitrary precision.
+func truncateMilliseconds(
+	nanoSeconds int, precision PrecisionLevel,
+) (millisecondsStr string) {
+	milliseconds := nanoSeconds / 1000
+	millisecondsStr = fmt.Sprintf("%06d", milliseconds)
+	millisecondsStr = millisecondsStr[:6-(PrecisionFull-precision)]
+
+	return millisecondsStr
+}
+
+// Const for time.Time timezone. We don't want a timezone because we don't REALLY know
+// that a TM or DA value is UTC, so we don't want UTC to be made explicit.
```

Comment:
The timezone comes from another tag right?

Reply:
Yes, timezone often comes from another tag, if it is defined at all for a given time value.